### DISCUSSION TITLE
Removing edges in BarGraphItem

### DISCUSSION THREAD

**Discussion by RT-Tap** [pyqtgraph/pyqtgraph#1789] [Q&A] [answered]
Hello, I'm faily new to pyqtgraph and am working on a little project and was hoping to get some help for a couple issues I've been facing, just one for right now, the others i should be able to iron out myself.  

So the main issue I have at the moment is trying to get rid of edges on bars in a BarGraphItem.  I would like to overlay one bar graph on top of another and was going with the following approach; create a GraphicsLayoutWidget then add a PlotItem and then add BarGraphItem into this view box and then add another barGraphItem on top with an alpha channel of around 50 in order to see through the top bar graph.  I got the infill/brush to be transparent but the edges I can't seem to change, is there anyway to do this?  I couldn't find anything for BarGraphItem specifically; I did find `edgecolor="none"` for bargraphs created (i assume) by using `pyqtgraph.plot(x,y,edgecolor="none") `  So any help would be greatly appreciated.  

Also I keep getting a message saying the rotate method for rotating BarGraphItem's and I assume other types of plots will be removed in 0.13, what is the alternate method of rotating graphs that i should be using?

**Comment by j9ac9k** [comment, marked as answer]
BarGraphItem definitely doesn't get as much love as some of our other plots; you want to set the `pen` object to something that's transparent, say `(0, 0, 0, 0)`; and see if that does the trick for you.  

The rotate method in `Qt.py` is being deprecated, you can see that here:

https://github.com/pyqtgraph/pyqtgraph/blob/master/pyqtgraph/Qt.py#L232-L240

You can see the transform that will be happening there in the function; you can specify vertical or horizontal bars by specifying x, or y along with width or height, and it should draw accordingly.

**Reply by RT-Tap** [reply]
You the best! Thank you very much!

**Reply by j9ac9k** [reply]
glad it worked out for you!

**Comment by RT-Tap** [comment]
Hey, I don't want to start a new thread for this but what is the best way to update a BarGraphItem?
At the moment I have an update function that changes just the height (maybe more later) and I'm calling it through a QtTimer that should and does call it every 100mS however I can't get it to update.
Things I've tried so far:
recreating the BarGraphItem using the same variable name and calling pg.QApplication.processEvents()
recreating the BarGraphItem, using the same variable name, removing it from viewbox then adding it back to viewBox.
creating a different BarGraphItem, removing old one from ViewBox and adding new one to ViewBox.
creating a different BarGraphItem, removing old one from ViewBox and adding new one to ViewBox then calling processEvents()

Ideally, as always something with the least overhead would be appreciated but I'll take anything to just learn and understand what direction I should be headed in.

Thank again and in advance!

EDIT: Sorry I realized this doesn't clarify anything at all and will update with code the first chance I get
